[Bug Fix] Release v1.11.0 Cleanup

harish876 opened the pull request:
Resolves:

 - [Issue 187](https://github.com/apache/incubator-resilientdb/issues/187)
 - [Issue 188](https://github.com/apache/incubator-resilientdb/issues/188)
 - [Issue 189](https://github.com/apache/incubator-resilientdb/issues/189)

cjcchen commented on tools/generate_region_config_cc.cpp:
what is this file?

harish876 replied:
the generate config script does not work. replaced the python version with a cpl equivalent

cjcchen commented:
Which config is generated dynamically?

On Sat, Nov 1, 2025 at 2:01 PM Harish ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In .licenserc.yaml
> <https://github.com/apache/incubator-resilientdb/pull/192#discussion_r2483142446>
> :
>
> > @@ -7,6 +7,10 @@ header:
>      - '.*'
>      - '.**/**'
>      - '**/*.sol'
> +    - '**/*.config'
>
> the config is generated dynamically. headers cannot be added to these.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-resilientdb/pull/192#discussion_r2483142446>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4YYLWVRN5H23ZGL53RX5332REC3AVCNFSM6AAAAACK2W3ZOCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTIMBWHA4DKMZXHA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

harish876 commented:
Answer to this: "Which config is generated dynamically?"

According to the README.md

## Custom Ports ##
When starting the service locally, current services are running on 10000 port-base with 5 services where the server config is located [here](https://github.com/apache/incubator-resilientdb/blob/master/service/tools/config/server/server.config)

If you want to change the setting,  you need to generate the certificates.

Go the the workspace where the resilientdb repo is localted.

Change the setting parameters here and run the script:
  ```shell
  ./service/tools/kv/server_tools/generate_config.sh
  ```

Then re-run the start script:
  ```shell
  ./service/tools/kv/server_tools/start_kv_service.sh
  ```
  
  
# Config Files Generated by generate_config.sh

## Generated Files

### 1. `server.config`
- **Created at**: line 52
- **Location**: remains in `${output_path}` directory
- **Purpose**: used as input to generate JSON configuration

### 2. `client.config`
- **Created at**: line 51
- **Moved to**: `${output_path}/interface/service.config` (line 110)
- **Purpose**: client configuration file

### 3. `${output_path}/server/server.config`
- **Source**: moved from `server.config.json` (line 109)
- **Generation**: JSON is generated from `server.config` at lines 102-104
- **Purpose**: JSON-based server configuration

## Summary of Final .config Files

1. `server.config` (in `${output_path}` directory)
2. `${output_path}/server/server.config` (JSON-based config)
3. `${output_path}/interface/service.config` (client config, moved from `client.config`)

## Notes

- `server.config` stays local in the output directory
- `client.config` is moved to `interface/service.config`
- The JSON-based server config is placed in `server/server.config`

cjcchen commented:
So remove the generated files

On Sun, Nov 2, 2025 at 12:14 PM Harish ***@***.***> wrote:

> *harish876* left a comment (apache/incubator-resilientdb#192)
> <https://github.com/apache/incubator-resilientdb/pull/192#issuecomment-3477430880>
>
> Answer to this: "Which config is generated dynamically?"
>
> According to the README.md
> Custom Ports
>
> When starting the service locally, current services are running on 10000
> port-base with 5 services where the server config is located here
> <https://github.com/apache/incubator-resilientdb/blob/master/service/tools/config/server/server.config>
>
> If you want to change the setting, you need to generate the certificates.
>
> Go the the workspace where the resilientdb repo is localted.
>
> Change the setting parameters here and run the script:
>
> ./service/tools/kv/server_tools/generate_config.sh
>
> Then re-run the start script:
>
> ./service/tools/kv/server_tools/start_kv_service.sh
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-resilientdb/pull/192#issuecomment-3477430880>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4YYLRZPSPVDBQV7YC23VD32WAITAVCNFSM6AAAAACK2W3ZOCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTINZXGQZTAOBYGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

harish876 commented:
Closing this for a bit. Will update soon
